### DISCUSSION TITLE
Is it possible to override a property value as part of a mapperly method?

### DISCUSSION THREAD

**Discussion by baynezy** [riok/mapperly#1417] [General]
Lets say I have the following classes:

**User**

```csharp
public class User
{
    public User(Guid id) : this(
        id, 
        DateTime.UtcNow, 
        DateTime.UtcNow)
    {
    }

    public User(
        Guid id, 
        DateTime createdAt, 
        DateTime updatedAt)
    {
        Id = id;
        CreatedAt = createdAt;
        UpdatedAt = updatedAt;
    }
    
    public Guid Id { get; }
    public required string Name { get; set; }
    public DateTime CreatedAt { get;  }
    public DateTime UpdatedAt { get;  }
}
```

**UserDto**

```csharp
public class UserDto
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public DateTime UpdatedAt { get; set; }
}
```

If I wanted to map from `UserDto` to `User`. Is there a way that I can provide the `CreatedAt` property from another source?

So I can have something like `public partial User ToUser(UserDto dto, DateTime createdAt);`

I cannot use [before and after map](https://mapperly.riok.app/docs/configuration/before-after-map/) as the only way to provide it is to the constructor.

**Comment by latonz** [comment]
This is supported since 4.0.0-next.2, see also https://github.com/riok/mapperly/issues/103.
